### PR TITLE
Fix the renderer circular reference

### PR DESCRIPTION
**Pull request by Connicpu**

### CODE REVIEW

File: src/directx/DXRenderer.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 #include "Windows.h"
 #include "../main/engine/Engine.h"
+#include "../main/engine/Renderer.h"
```
Comment (Connicpu) [inline src/directx/DXRenderer.h:9]: We now have to include this header here since Engine.h does not

File: src/main/engine/Renderer.h [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-#include "Engine.h"
+class Engine;
```
Comment (Connicpu) [inline src/main/engine/Renderer.h:6]: We don't actually need to know about the contents of the Engine class from inside this header.

File: src/directx/DXRenderer.cpp [hunk 3]
```diff
@@ -72,7 +72,7 @@ void DXRenderer::Draw()
 	// TODO: Draw into backbuffer, then present it
 }
 
-LRESULT CALLBACK DXRenderer::StaticWindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+LRESULT DXRenderer::StaticWindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
```
Comment (Connicpu) [inline src/directx/DXRenderer.cpp:75]: Calling conventions only need to be specified in the header

**Comment by berwyn** [comment]
Thanks for pointing out the calling convention bit! That's something I hadn't known :D

**Pull request merged**